pkg/webapi/webapi.go

```diff
@@ -143,6 +150,26 @@ func (srv *server) handleState(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	srv.stopTimerMu.Lock()
+	srv.stopTimer.Stop()
+	srv.stopTimerMu.Unlock()
+
+	defer func() {
```

Comment:
shouldn't this be a go-routine? as it is now, it is called when the for-loop on line 194 terminates, which it only does when an error is thrown

Reply:
The purpose of this is to send a stop command to TRC after `timeout` of inactivity. This function is called once the `handleState` returns, hence, when the WebSocket connection is closed.
Once the WS is closed, a timer is started(L161), once it expires, the "stop" function is called. The timer can stopped on L154 by a new WebSocket connection, if it's open within the `timeout`.